NorthstarDLL/version.cpp

```diff
@@ -1,26 +1,94 @@
 #include "version.h"
 #include "pch.h"
 #include "ns_version.h"
+#include "dedicated.h"
 
 char version[16];
-char NSUserAgent[32];
+char NSUserAgent[256];
 
 void InitialiseVersion()
 {
 	constexpr int northstar_version[4] {NORTHSTAR_VERSION};
+	int ua_len = 0;
 
 	// We actually use the rightmost integer do determine whether or not we're a debug/dev build
 	// If it is set to 1, we are a dev build
 	// On github CI, we set this 1 to a 0 automatically as we replace the 0,0,0,1 with the real version number
 	if (northstar_version[3] == 1)
 	{
 		sprintf(version, "%d.%d.%d.%d+dev", northstar_version[0], northstar_version[1], northstar_version[2], northstar_version[3]);
-		sprintf(NSUserAgent, "R2Northstar/%d.%d.%d+dev", northstar_version[0], northstar_version[1], northstar_version[2]);
+		ua_len += snprintf(
+			NSUserAgent + ua_len,
+			sizeof(NSUserAgent) - ua_len,
+			"R2Northstar/%d.%d.%d+dev",
+			northstar_version[0],
+			northstar_version[1],
+			northstar_version[2]);
 	}
 	else
 	{
 		sprintf(version, "%d.%d.%d.%d", northstar_version[0], northstar_version[1], northstar_version[2], northstar_version[3]);
-		sprintf(NSUserAgent, "R2Northstar/%d.%d.%d", northstar_version[0], northstar_version[1], northstar_version[2]);
+		ua_len += snprintf(
+			NSUserAgent + ua_len,
+			sizeof(NSUserAgent) - ua_len,
+			"R2Northstar/%d.%d.%d",
+			northstar_version[0],
+			northstar_version[1],
+			northstar_version[2]);
 	}
+
+	if (IsDedicatedServer())
+		ua_len += snprintf(NSUserAgent + ua_len, sizeof(NSUserAgent) - ua_len, " (Dedicated)");
+
+	// ntdll will always be loaded
+	HMODULE ntdll = GetModuleHandleA("ntdll");
```

Comment:
I'd add a comment somewhere around here that the following code block is about getting Windows and/or Wine version.

Reply:
Done.